draco/synthesis/stream.py

```diff
@@ -243,6 +246,162 @@ def process(self, sstream):
 
 
 class MakeTimeStream(task.SingleTask):
+    """Generate a time stream from a sidereal stream.
+
+    Requires an input dataset with a time axis to mimic.
+
+    Attributes
+    ----------
+    lanczos_width : int
+        Width of the Lanczos interpolation kernel.
+    """
+
+    lanczos_width = config.Property(proptype=int, default=5)
+
+    def setup(self, observer):
+        """Set up a telescope object.
+
+        Parameters
+        ----------
+        observer : ProductManager
+            Telescope manager
+        """
+        self.observer = io.get_telescope(observer)
+
+    def process(self, sstream, tstream):
+        """Sample the sidereal stream at a set of times.
+
+        Parameters
+        ----------
+        sstream : containers.SiderealStream | containers.HybridVisStream
+            Dataset sampled in RA
+        tstream : containers.TODContainer | HybridVisStream
+            Dataset sampled in time
+
+        Returns
+        -------
+        timestream : containers.TimeStream | containers.HybridVisStream
+            sstream data sampled in time
+        """
+        # Figure out the time-like axis of `tstream`
+        if hasattr(tstream, "time"):
+            time = tstream.time[:]
+            tra = self.observer.unix_to_lsa(time)
+        elif hasattr(tstream, "ra"):
+            tra = tstream.ra[:]
+            lsd = tstream.attrs.get("lsd", tstream.attrs.get("csd"))
+            time = self.observer.lsd_to_unix(lsd + tra / 360.0)
+
+        # Determine output container based on input container
+        container_map = {
+            containers.SiderealStream: containers.TimeStream,
+            containers.HybridVisStream: containers.HybridVisStream,
+        }
+
+        # Make the new timestream containers
+        for cls in inspect.getmro(sstream.__class__):
+            OutputContainer = container_map.get(cls)
+
+            if OutputContainer is not None:
+                break
+
+        if OutputContainer is None:
+            raise TypeError(
+                f"No valid container mapping.\nGot {sstream.__class__}.\n"
+                f"Mappings exist for {list(container_map.keys())}."
+            )
+
+        # Provide both the time and ra axes - whichever one isn't
+        # used will be ignored
+        out = OutputContainer(axes_from=sstream, attrs_from=sstream, time=time, ra=tra)
+
+        # Make sure datasets are distributed correctly
+        sstream.redistribute("freq")
+        out.redistribute("freq")
+
+        # Make the interpolation array
+        R = regrid.lanczos_forward_matrix(sstream.ra, tra, self.lanczos_width).T.copy()
```

Comment:
```suggestion
        R = regrid.lanczos_forward_matrix(sstream.ra, tra, self.lanczos_width, periodic=True).T.copy()
```

Reply:
For a `HybridVisStream` generated directly from a `TimeStream`, if the time axis extends beyond the bounds of a sidereal day, it will take on values that are either negative or greater than 360.  This ensures that lines like:
```
time = self.observer.lsd_to_unix(lsd + tra / 360.0)
```
still produce the correct time. So, I believe that in addition to setting `periodic = True`,  we also need to wrap `tra` between 0 and 360 before passing it to this task, i.e., `tra % 360.0`.